util/gosdk/instance.go

```diff
@@ -0,0 +1,49 @@
+package gosdk
+
+import (
+	"context"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/linode/cluster-api-provider-linode/util/logging"
+	"github.com/linode/linodego"
+)
+
+// InstanceHandler contains helpers to handle Linode instances.
+type InstanceHandler interface {
+	CreateInstance(context.Context, linodego.InstanceCreateOptions) (*linodego.Instance, error)
+}
+
+// NewInstanceHandler constructs a Linode instance handler.
+func NewInstanceHandler(client *Client, log logr.Logger, createTimeout time.Duration) InstanceHandler {
+	return &instanceHandler{
+		client:        client,
+		createTimeout: int(createTimeout.Round(time.Second)),
+		log:           log.WithName("instance-handler"),
+	}
+}
+
+type instanceHandler struct {
+	client        *Client
+	createTimeout int
+	log           logr.Logger
+}
+
+// CreateInstance creates an instance and wits for booting.
+func (ih *instanceHandler) CreateInstance(ctx context.Context, instanceOptions linodego.InstanceCreateOptions) (*linodego.Instance, error) {
+	instance, err := ih.client.CreateInstance(ctx, instanceOptions)
+	if err != nil {
+		return nil, logging.LogAndWrapError(ih.log, "instance creation failed", err, instanceOptions)
+	}
+
+	ih.log.V(3).Info("instance created", "ID", instance.ID)
+
+	instance, err = ih.client.WaitForInstanceStatus(ctx, instance.ID, linodego.InstanceRunning, ih.createTimeout)
+	if err != nil {
+		return nil, logging.LogAndWrapError(ih.log, "wait for instance creation failed", err, instanceOptions)
+	}
+
+	ih.log.V(3).Info("instance running", "ID", instance.ID)
```

Comment:
Should we not to block the overall process by waiting for the status of an instance?

Reply:
@zliang-akamai The reason for the block is machine controller has to know when an instance is ready to deploy Kubernetes. So i think at some point we have to know the machine status to update custom resource status. May i ask what is in your head about status updates?

Reply:
i dont think we should be blocking anything in a reconcile loop, should be setting it to create and requeue the machine to run again in 5s and check the status until it's ready. No point creating our own blocking methods when controller framework was designed to take care of that for us

Reply:
@luthermonson My plan was also to not block reconciliation loop, I just experienced higher CPU load during re-queuing items, than retrying or waiting things in a separated goroutine. Of course we pay with memory on the other hand. We have fixed number of reconciliation loops, so if we have lot's of re-queues, we should increase reaction time. On the other hand we can create millions of sleeping goroutines. WDYT?